diff --git "a/N:\\temp\\msc\\marc2025.1\\tools\\include_linux64.orig" "b/N:\\temp\\msc\\marc2025.1\\tools\\include_linux64"
index 8318de6..8472d71 100644
--- "a/N:\\temp\\msc\\marc2025.1\\tools\\include_linux64.orig"
+++ "b/N:\\temp\\msc\\marc2025.1\\tools\\include_linux64"
@@ -208,6 +208,15 @@ if test -n "$MSCCOSIM_HOME"; then
  MARC_COSIM_LIB="$MSCCOSIM_HOME/CoSim$MSCCOSIM_VERSION/Dcosim$MSCCOSIM_VERSION/lib"
 fi
 
+# DAMASK uses the HDF5 compiler wrapper around the Intel compiler
+H5FC=$(h5fc -shlib -show)
+if [[ "$H5FC" == *"$dir is"* ]]; then
+  H5FC=$(echo $(echo "$H5FC" | tail -n1) | sed -e "s/\-shlib/-fPIC -qopenmp/g")
+  H5FC=${H5FC%-lmpifort*}
+fi
+HDF5_LIB=${H5FC//*ifort/}
+FCOMP="$H5FC"
+
 # AEM
 if test "$MARCDLLOUTDIR" = ""; then
  DLLOUTDIR="$MARC_LIB"
@@ -669,6 +678,11 @@ CCHIGH="icx -c $CC_OPT -O3 $I8DEFINES -DLinux -DLINUX -DLinux_intel $CDEFINES $C
 CC_TETMESH="icpx -c -O2 -I$TETMESHDIR/include -qopenmp -ansi-alias -m64 -pipe -fPIC -falign-functions=16 -fstrict-aliasing -fno-jump-tables -MMD -DLINUX -D_CRT_SECURE_NO_WARNINGS -DX64 -DMESHINGAPI_LIB -DNDEBUG"
 CC_CAEFATIGUE="icpx -c -O2 -I$CAEFATIGUEDIR/linux64/include -qopenmp -ansi-alias -m64 -pipe -fPIC -falign-functions=16 -fstrict-aliasing -fno-jump-tables -MMD -DLINUX -D_CRT_SECURE_NO_WARNINGS -DX64 -DNDEBUG"
 
+# DAMASK compiler calls
+DCCLOW="icx -c $CC_OPT -O0 $I8DEFINES -DLinux -DLINUX -DLinux_intel -DFYAML -DMARC_SOURCE $CDEFINES $CINCL $SOLVERFLAGS $OPENSSL_INCLUDE "
+DCC="icx -c $CC_OPT -O1 $I8DEFINES -DLinux -DLINUX -DLinux_intel -DFYAML -DMARC_SOURCE $CDEFINES $CINCL $SOLVERFLAGS $OPENSSL_INCLUDE -I$MPI_ROOT/include"
+DCCHIGH="icx -c $CC_OPT -O3 $I8DEFINES -DLinux -DLINUX -DLinux_intel -DFYAML -DMARC_SOURCE $CDEFINES $CINCL $SOLVERFLAGS $OPENSSL_INCLUDE "
+
 if test "$MARCDEBUG" = "ON"
 then
   CC="icx -c $CC_OPT -DLinux $I8DEFINES -DLINUX -DLinux_intel $CDEFINES $CINCL $SOLVERFLAGS $OPENSSL_INCLUDE -I$MPI_ROOT/include "
@@ -699,7 +713,7 @@ then
 PROFILE=" $PROFILE -pg"
 fi
 
-FORT_OPT="-c -assume byterecl -safe-cray-ptr -mp1 -WB -fp-model source -diag-disable:10448"
+FORT_OPT="-c -implicitnone -stand f18 -standard-semantics -assume nostd_mod_proc_name  -safe-cray-ptr -mp1 -WB -fp-model source -diag-disable:10448"
 # FORT_OPT="-c -assume byterecl -safe-cray-ptr -mp1 -WB "
 if test "$MTHREAD" = "OPENMP"
 then
@@ -713,7 +727,7 @@ else
   FORT_OPT=" $FORT_OPT -save -zero"
 fi
 if test "$MARCHDF_HDF" = "HDF"; then
-  FORT_OPT="$FORT_OPT -DMARCHDF_HDF=$MARCHDF_HDF $HDF_INCLUDE"
+  FORT_OPT="$FORT_OPT -DMARCHDF=$MARCHDF_HDF"
 fi
 if test "$NASTRAN_LIB" = "on" ; then
   FORT_OPT="$FORT_OPT -fPIC "
@@ -730,6 +744,41 @@ FORTNA="$FCOMP $FORT_OPT -fno-alias -O3 $I8FFLAGS -I$MARC_SOURCE/common \
 #  for compiling free form f90 files. high opt, integer(4)
 FORTF90="$FCOMP -c -O3"
 
+# determine DAMASK version
+if test -n "$DAMASK_USER"; then
+  DAMASKROOT=`dirname $DAMASK_USER`/../..
+  read DAMASK_VERSION < $DAMASKROOT/VERSION
+  DAMASK_VERSION_MAJOR="'`echo $DAMASK_VERSION | cut -d. -f1`'"
+  DAMASK_VERSION_MINOR="'`echo $DAMASK_VERSION | cut -d. -f2`'"
+  DAMASK_VERSION_PATCH=`echo $DAMASK_VERSION | cut -d. -f3`
+  if [[ $DAMASK_VERSION_PATCH == *"-"* ]]; then
+   DAMASK_VERSION_HASH="'`echo $DAMASK_VERSION_PATCH | cut -d- -f3`'"
+  fi
+  DAMASK_VERSION_PATCH="'`echo $DAMASK_VERSION_PATCH | cut -d- -f1`'"
+  DAMASK_VERSION="'"$DAMASK_VERSION"'"
+else
+  DAMASK_VERSION="'N/A'"
+fi
+if [[ -n $DAMASK_VERSION_HASH ]]; then
+  DAMASK_INFO="-DDAMASKVERSION=$DAMASK_VERSION -DDAMASK_VERSION_MAJOR=$DAMASK_VERSION_MAJOR -DDAMASK_VERSION_MINOR=$DAMASK_VERSION_MINOR -DDAMASK_VERSION_PATCH=$DAMASK_VERSION_PATCH -DDAMASK_VERSION_HASH=$DAMASK_VERSION_HASH"
+else
+  DAMASK_INFO="-DDAMASKVERSION=$DAMASK_VERSION -DDAMASK_VERSION_MAJOR=$DAMASK_VERSION_MAJOR -DDAMASK_VERSION_MINOR=$DAMASK_VERSION_MINOR -DDAMASK_VERSION_PATCH=$DAMASK_VERSION_PATCH"
+fi
+
+# DAMASK compiler calls
+DFORTLOWMP="$FCOMP -c -O0 -qno-offload -implicitnone -stand f18 -standard-semantics -assume nostd_mod_proc_name -safe-cray-ptr $PROFILE -zero -mp1 -WB $I8FFLAGS -I$MARC_SOURCE/common \
+ -fpp -ftz -diag-disable 5268 -warn declarations -warn general -warn usage -warn interfaces -warn ignore_loc -warn alignments -DFYAML -DMARC_SOURCE=$MARC_SOURCE $DAMASK_INFO \
+ -qopenmp -qopenmp-threadprivate=compat\
+  $MUMPS_INCLUDE $I8DEFINES -DLinux -DLINUX -DLinux_intel $FDEFINES $DDM $SOLVERFLAGS -I$KDTREE2_MOD -I$MARC_MOD"
+DFORTRANMP="$FCOMP -c -O1 -qno-offload -implicitnone -stand f18 -standard-semantics -assume nostd_mod_proc_name -safe-cray-ptr $PROFILE -zero -mp1 -WB $I8FFLAGS -I$MARC_SOURCE/common \
+ -fpp -ftz -diag-disable 5268 -warn declarations -warn general -warn usage -warn interfaces -warn ignore_loc -warn alignments -DMARC_SOURCE=$MARC_SOURCE $DAMASK_INFO \
+ -qopenmp -qopenmp-threadprivate=compat\
+  $MUMPS_INCLUDE $I8DEFINES -DLinux -DLINUX -DLinux_intel $FDEFINES $DDM $SOLVERFLAGS -I$KDTREE2_MOD -I$MARC_MOD"
+DFORTHIGHMP="$FCOMP -c -O3 -qno-offload -implicitnone -stand f18 -standard-semantics -assume nostd_mod_proc_name -safe-cray-ptr $PROFILE -zero -mp1 -WB $I8FFLAGS -I$MARC_SOURCE/common \
+ -fpp -ftz -diag-disable 5268 -warn declarations -warn general -warn usage -warn interfaces -warn ignore_loc -warn alignments -DMARC_SOURCE=$MARC_SOURCE $DAMASK_INFO \
+ -qopenmp -qopenmp-threadprivate=compat\
+  $MUMPS_INCLUDE $I8DEFINES -DLinux -DLINUX -DLinux_intel $FDEFINES $DDM $SOLVERFLAGS -I$KDTREE2_MOD -I$MARC_MOD"
+
 if test "$MARCDEBUG" = "ON"
 then
   FORTLOW="$FCOMP $FORT_OPT $PROFILE $I8FFLAGS -I$MARC_SOURCE/common \
@@ -919,7 +968,7 @@ SOLVERLIBS="${BCSSOLVERLIBS} ${VKISOLVERLIBS} ${CASISOLVERLIBS} ${MF2SOLVERLIBS}
   -lmg-hybrid \
   -lmg-cadsurf \
   -lmg-hexa \
-  $HDF_LIBS \
+  $HDF5_LIB \
   $SOLVER2LIBS \
   $RTREE_LIB/librtree_lib.a \
   $SFGEO_LIB/libgeo_lib.a"
